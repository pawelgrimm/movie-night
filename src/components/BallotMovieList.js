import React, { useEffect, useState } from "react";
import { connect } from "react-redux";
import { useParams } from "react-router-dom";
import MovieListItem from "./MovieListItem";
import { ThumbsUp, ThumbsDown } from "react-feather";
import { openModal } from "../actions/videoModal";
import { getMoviesOnBallot } from "../firebase/api";
import Loader from "./Loader";
import { hydrateMovies } from "../selectors/movie";
import { fetchMovie } from "../actions/movie";

const mapStateToProps = ({ movies }) => ({
  savedMovies: movies,
});

const mapDispatchToProps = (dispatch) => ({
  openModal: (videos) => dispatch(openModal(videos)),
  fetchMovie: (id) => dispatch(fetchMovie(id)),
});

export const BallotMovieList = ({ savedMovies, openModal, fetchMovie }) => {
  const [loaded, setLoaded] = useState(false);
  const [movies, setMovies] = useState([]);
  const [movieIds, setMovieIds] = useState([]);
  const { id } = useParams();

  useEffect(() => {
    getMoviesOnBallot(id).then((ref) => {
      const movieIds = ref.val();
      movieIds.forEach((id) => fetchMovie(id));
      setMovieIds(movieIds);
      setMovies(hydrateMovies(savedMovies, movieIds));
      setLoaded(true);
    });
  }, []);

  useEffect(() => {
    if (loaded) {
      setMovies(hydrateMovies(savedMovies, movieIds));
    }
  }, [savedMovies]);

  if (!loaded) {
    return <Loader />;
  }

  return (
    <div className="list-body">
      {movies.map((movie) => {
        const id = movie.info.id;
        return (
          <MovieListItem
            key={id}
            isLoading={!movie.isSaved}
            movie={movie.info}
            collapsedButtons={collapsedButtons(id)}
            expandedButtons={expandedButtons(
              id,
              movie.info.trailers,
              openModal
            )}
          />
        );
      })}
    </div>
  );
};

const collapsedButtons = (movieId) => makeButtons(movieId, true);

const expandedButtons = (movieId, videos, openModal) => [
  videos && videos.length > 0 && (
    <button
      className="button"
      onClick={(event) => {
        event.stopPropagation();
        openModal(videos);
      }}
    >
      Watch Trailer
    </button>
  ),
  ...makeButtons(movieId, false),
];

const makeButtons = (movieId, collapsed = true) => [
  <button
    className="button"
    onClick={(event) => {
      event.stopPropagation();
      console.log("thumbs up clicked for", movieId);
    }}
  >
    <ThumbsUp />
  </button>,
  <button
    className="button"
    onClick={(event) => {
      event.stopPropagation();
      console.log("thumbs up clicked for", movieId);
    }}
  >
    <ThumbsDown />
  </button>,
];

export default connect(mapStateToProps, mapDispatchToProps)(BallotMovieList);
