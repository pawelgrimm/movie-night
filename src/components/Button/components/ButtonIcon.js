import React from "react";

const ButtonIcon = ({ children, ...rest }) => {
  return <>{children}</>;
};

export default ButtonIcon;
