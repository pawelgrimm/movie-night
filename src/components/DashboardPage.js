import React, {useReducer} from "react";
import MovieSearch from "./MovieSearch";
import searchReducer from "../reducers/search";
import SearchContext from "../context/search-context";

const DashboardPage = () => {
    const [ searchResults, dispatch ] = useReducer(searchReducer, []);
    const setBackground = (imageUrl) => {

    }
    return (
        <SearchContext.Provider value={{searchResults, dispatch}}>
            <div className="content-container">
                <MovieSearch />
            </div>
        </SearchContext.Provider>
    );
}

export default DashboardPage;
