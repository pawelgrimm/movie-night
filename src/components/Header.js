import React from "react";
import { Link } from "react-router-dom";

export const Header = () => (
  <header className="header">
    <div className="content-container">
      <div className="header__content">
        <Link
          className="header__title"
          to="/dashboard"
        >
          <h1>Movie Night</h1>
        </Link>
      </div>
    </div>
  </header>
);

export default Header;
