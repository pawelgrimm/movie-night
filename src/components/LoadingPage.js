import React from "react";
import Loader from "./Loader";

const LoadingPage = () => (
  <div className="loader">
    <Loader />
  </div>
);

export default LoadingPage;
