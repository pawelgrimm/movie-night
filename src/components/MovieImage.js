import ReactImageFallback from "react-image-fallback";
import React from "react";
import { getImageUrl } from "../theMovieDb/theMovieDb";

const MovieImage = ({ title, type, imageWidth, filePath, className = "" }) => {
  const imagePath = getImageUrl(type, imageWidth, filePath);
  return (
    <ReactImageFallback
      src={imagePath}
      fallbackImage="/images/default_poster.png"
      initialImage="/images/loader.gif"
      alt={`${type} image for ${title}`}
      className={className || type + "-image"}
    />
  );
};

export default MovieImage;
