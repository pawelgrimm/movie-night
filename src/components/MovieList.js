import React from "react";
import { connect } from "react-redux";
import MovieListItem from "./MovieListItem";

const mapStateToProps = ({ movies }) => ({ movies });

export const MovieList = ({ movies, toggleModal }) => {
  return (
    <div className="list-body">
      {movies.length > 0 ? (
        movies.map((movie) => (
          <MovieListItem
            key={movie.id}
            movie={movie}
            toggleModal={toggleModal}
          />
        ))
      ) : (
        <div>
          <h3>Welcome to Movie Night!</h3>
          <p>Use the search to add some movies to your list.</p>
          <p>
            Once you're done, send the list to your friends. They will be able
            to vote on the movies you selected!
          </p>
          <p>
            When all the votes are in, the most favorable movie will be selected
            for your movie night!
          </p>
          <p>Enjoy!</p>
        </div>
      )}
    </div>
  );
};

export default connect(mapStateToProps)(MovieList);
