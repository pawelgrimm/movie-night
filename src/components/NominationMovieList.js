import React, { useEffect, useState } from "react";
import { connect } from "react-redux";
import MovieListItem from "./MovieListItem";
import { Trash2 as TrashIcon } from "react-feather";
import { fetchMovie } from "../actions/movie";
import { openModal } from "../actions/videoModal";
import { hydrateMovies } from "../selectors/movie";
import { removeMovie } from "../actions/ballot";
import Loader from "./Loader";

const mapStateToProps = ({ movies, ballot }) => ({
  movies: hydrateMovies(movies, ballot.movies),
});

const mapDispatchToProps = (dispatch) => ({
  fetchMovie: (id) => dispatch(fetchMovie(id)),
  removeMovie: (id) => dispatch(removeMovie(id)),
  openModal: (videos) => dispatch(openModal(videos)),
});

export const NominationMovieList = ({
  movies,
  openModal,
  removeMovie,
  fetchMovie,
}) => {
  return (
    <div className="list-body">
      {movies.map((movie) => {
        const id = movie.info.id;
        fetchMovie(id);
        return (
          <MovieListItem
            key={id}
            isLoading={!movie.isSaved}
            movie={movie.info}
            collapsedButtons={collapsedButtons(id, removeMovie)}
            expandedButtons={expandedButtons(
              id,
              movie.info.trailers,
              openModal,
              removeMovie
            )}
          />
        );
      })}
    </div>
  );
};

export default connect(
  mapStateToProps,
  mapDispatchToProps
)(NominationMovieList);

const collapsedButtons = (movieId, removeMovie) => [
  <button
    className="button button--secondary movie-item__button--small"
    onClick={(event) => {
      event.stopPropagation();
      removeMovie(movieId);
    }}
  >
    <TrashIcon />
  </button>,
];

const expandedButtons = (movieId, videos, openModal, removeMovie) => [
  videos && videos.length > 0 && (
    <button
      className="button"
      onClick={(event) => {
        event.stopPropagation();
        openModal(videos);
      }}
    >
      Watch Trailer
    </button>
  ),
  <button
    className="button button--secondary"
    onClick={(event) => {
      event.stopPropagation();
      removeMovie(movieId);
    }}
  >
    Remove Movie
  </button>,
];
