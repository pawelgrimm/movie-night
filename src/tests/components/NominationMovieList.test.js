import { expect } from "@jest/globals";
import { shallow } from "enzyme";
import React from "react";
import { NominationMovieList } from "../../components/NominationMovieList";
import { movies } from "../fixtures/fixtures";

test("should render NominationMovieList correctly with movies", () => {
  const wrapper = shallow(<NominationMovieList movies={movies} />);
  expect(wrapper).toMatchSnapshot();
});

test("should render NominationMovieList correctly without movies", () => {
  const wrapper = shallow(<NominationMovieList movies={[]} />);
  expect(wrapper).toMatchSnapshot();
});
