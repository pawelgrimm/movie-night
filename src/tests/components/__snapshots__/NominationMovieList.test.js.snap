// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should render NominationMovieList correctly with movies 1`] = `
<div
  className="list-body"
>
  <Connect(MovieListItem)
    collapsedButtons={
      Array [
        <button
          className="button button--secondary movie-item__button--small"
          onClick={[Function]}
        >
          <Trash2 />
        </button>,
      ]
    }
    expandedButtons={
      Array [
        undefined,
        <button
          className="button button--secondary"
          onClick={[Function]}
        >
          Remove Movie
        </button>,
      ]
    }
    key="132344"
    movie={
      Object {
        "adult": false,
        "backdrop_path": "/lv9UT4VyoyGFTKNBboMTeZ5ZcD3.jpg",
        "genre_ids": Array [
          18,
          10749,
        ],
        "id": 132344,
        "original_language": "en",
        "original_title": "Before Midnight",
        "overview": "We meet Jesse and Celine nine years on in Greece. Almost two decades have passed since their first meeting on that train bound for Vienna.",
        "popularity": 15.97,
        "poster_path": "/qbGKJmNUroDz75kh5Oafoall89e.jpg",
        "release_date": "2013-04-05",
        "title": "Before Midnight",
        "video": false,
        "vote_average": 7.5,
        "vote_count": 1351,
      }
    }
  />
  <Connect(MovieListItem)
    collapsedButtons={
      Array [
        <button
          className="button button--secondary movie-item__button--small"
          onClick={[Function]}
        >
          <Trash2 />
        </button>,
      ]
    }
    expandedButtons={
      Array [
        undefined,
        <button
          className="button button--secondary"
          onClick={[Function]}
        >
          Remove Movie
        </button>,
      ]
    }
    key="76"
    movie={
      Object {
        "adult": false,
        "backdrop_path": "/qA2TyqPldTtoTVY3LKrNIG5g6bH.jpg",
        "genre_ids": Array [
          18,
          10749,
        ],
        "id": 76,
        "original_language": "en",
        "original_title": "Before Sunrise",
        "overview": "On his way to Vienna, American Jesse meets CÃ©line, a student returning to Paris. After long conversations forge a surprising connection between them, Jesse convinces Celine to get off the train with him in Vienna. Since his flight to the U.S. departs the next morning and he has no money for lodging, they wander the city together, taking in the experiences of Vienna and each other.",
        "popularity": 16.646,
        "poster_path": "/khKJwHxgXSpDl3iKGJ5S15I8ABI.jpg",
        "release_date": "1995-01-27",
        "title": "Before Sunrise",
        "video": false,
        "vote_average": 7.9,
        "vote_count": 2292,
      }
    }
  />
  <Connect(MovieListItem)
    collapsedButtons={
      Array [
        <button
          className="button button--secondary movie-item__button--small"
          onClick={[Function]}
        >
          <Trash2 />
        </button>,
      ]
    }
    expandedButtons={
      Array [
        undefined,
        <button
          className="button button--secondary"
          onClick={[Function]}
        >
          Remove Movie
        </button>,
      ]
    }
    key="80"
    movie={
      Object {
        "adult": false,
        "backdrop_path": "/90MdQTVdjso5J35xPurGC6M2Tma.jpg",
        "genre_ids": Array [
          18,
          10749,
        ],
        "id": 80,
        "original_language": "en",
        "original_title": "Before Sunset",
        "overview": "Nine years ago two strangers met by chance and spent a night in Vienna that ended before sunrise. They are about to meet for the first time since. Now they have one afternoon to find out if they belong together.",
        "popularity": 13.65,
        "poster_path": "/gycdE1ARByGQcK4fYR2mgpU6OO.jpg",
        "release_date": "2004-02-10",
        "title": "Before Sunset",
        "video": false,
        "vote_average": 7.8,
        "vote_count": 1886,
      }
    }
  />
  <Connect(MovieListItem)
    collapsedButtons={
      Array [
        <button
          className="button button--secondary movie-item__button--small"
          onClick={[Function]}
        >
          <Trash2 />
        </button>,
      ]
    }
    expandedButtons={
      Array [
        undefined,
        <button
          className="button button--secondary"
          onClick={[Function]}
        >
          Remove Movie
        </button>,
      ]
    }
    key="43566"
    movie={
      Object {
        "adult": false,
        "backdrop_path": "/mnmYNA05niLp0Gb75UTIUvs8emo.jpg",
        "genre_ids": Array [
          18,
          9648,
          53,
        ],
        "id": 43566,
        "original_language": "en",
        "original_title": "Before and After",
        "overview": "Two parents deal with the effects when their son is accused of murdering his girlfriend.",
        "popularity": 12.741,
        "poster_path": "/4Q2SdpMJtNcbEvidNuc7z0KRiHo.jpg",
        "release_date": "1996-02-23",
        "title": "Before and After",
        "video": false,
        "vote_average": 6.1,
        "vote_count": 92,
      }
    }
  />
  <Connect(MovieListItem)
    collapsedButtons={
      Array [
        <button
          className="button button--secondary movie-item__button--small"
          onClick={[Function]}
        >
          <Trash2 />
        </button>,
      ]
    }
    expandedButtons={
      Array [
        undefined,
        <button
          className="button button--secondary"
          onClick={[Function]}
        >
          Remove Movie
        </button>,
      ]
    }
    key="296096"
    movie={
      Object {
        "adult": false,
        "backdrop_path": "/exYvOaUBplJUAqEB2M6QFgEF6Lp.jpg",
        "genre_ids": Array [
          18,
          10749,
        ],
        "id": 296096,
        "original_language": "en",
        "original_title": "Me Before You",
        "overview": "A small town girl is caught between dead-end jobs. A high-profile, successful man becomes wheelchair bound following an accident. The man decides his life is not worth living until the girl is hired for six months to be his new caretaker. Worlds apart and trapped together by circumstance, the two get off to a rocky start. But the girl becomes determined to prove to the man that life is worth living and as they embark on a series of adventures together, each finds their world changing in ways neither of them could begin to imagine.",
        "popularity": 18.085,
        "poster_path": "/oo2Q3T8juOYFh2G1YJ3oSMAfUyq.jpg",
        "release_date": "2016-06-01",
        "title": "Me Before You",
        "video": false,
        "vote_average": 7.8,
        "vote_count": 7377,
      }
    }
  />
</div>
`;

exports[`should render NominationMovieList correctly without movies 1`] = `
<div
  className="list-body"
>
  <div>
    <h3>
      Welcome to Movie Night!
    </h3>
    <p>
      Use the search to add some movies to your list.
    </p>
    <p>
      Once you're done, send the list to your friends. They will be able to vote on the movies you selected!
    </p>
    <p>
      When all the votes are in, the most favorable movie will be selected for your movie night!
    </p>
    <p>
      Enjoy!
    </p>
  </div>
</div>
`;
